### PR TITLE
Adding xrftomo

### PR DESCRIPTION
**Pull request by FabricioSMarin**
Checklist
- [x ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x ] Source is from official source
- [x ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x ] If static libraries are linked in, the license of the static library is packaged.
- [x ] Build number is 0
- [x ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by FabricioSMarin** [comment]
I ended up putting it as part of the version name in line 3, but I can change it to v{{ version}} instead.

Ironically, the check is failing on linux_64 which was the os it was developed on, but passing for win and osx which were the ones with the most peculiar bugs.

File: recipes/xrftomo/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  run:
+    - dxchange
+    - dxfile
+    - h5py
+    - matplotlib
+    - numpy
+    - pandas
+    - pyqt >=5
+    - pyqtgraph
+    - python >=3.7
```
Comment (carterbox) [inline recipes/xrftomo/meta.yaml]: You should not add a runtime requirement in this way. Do you really require `python>=3.7`? If not, then use `noarch` (ignore the python=2.7 it's already EOL?) If so, then you don't remove the `noarch` and use skip to skip python versions less than 3.7.

**Comment by carterbox** [comment]
Also, your Linux build is failing because pyQt5 cannot find the OpenGL library. I would ignore this. This is not a problem with your package.

File: recipes/xrftomo/meta.yaml (hunk 1 above)
Comment (carterbox) [reply recipes/xrftomo/meta.yaml]: You will need to rerender the feedstock whenever you change `skip` or `noarch`.

Comment (FabricioSMarin) [reply recipes/xrftomo/meta.yaml]: no, python >=3.7 is not absolutely necessary, but python>=2.x is. I am getting an error when using `noarch` and `skip: true [selector]` so it probably wouldn't like something like  `skip: true [py>2.x]` unless that is something that can be ignored as well. 

Aren't all the checks require to pass before conda-forge can accept my PR, if that's the case then how can the failing py27 or linux build be ignored? 

How do I rerender the feedstock? I thought that was only necessary until after the PR was approved.

Comment (carterbox) [reply recipes/xrftomo/meta.yaml]: My mistake. You were right the first time. You should add requirements on the Python version in `requirements/run` and `requirements/host`. Please follow this example: https://github.com/conda-forge/conda-forge.github.io/issues/486#issuecomment-344946793.

These pulls are merged by humans; you can ask whether you should expect the import of pyQt5 to work correctly on the feedstock. If not, you can disable that test for the Linux build.

To rerender, [ask the conda-forge-admin](https://conda-forge.readthedocs.io/en/latest/webservice.html) or use conda-smithy.

**Comment by carterbox** [comment]
@conda-forge-admin, please rerender

**Comment by npavlovikj** [comment]
That command works only in feedstock - you probably want to use `@conda-forge-admin, please restart ci` instead, https://github.com/conda-forge/conda-forge.github.io/blob/master/src/maintainer/infrastructure.rst#admin-web-services.

**Comment by carterbox** [comment]
@FabricioSMarin, you can make the OpenGL library available for the import test on Linux by [adding a yum_requirements.txt](https://conda-forge.org/docs/maintainer/knowledge_base.html#libgl) to `recipes/xrftomo`.

File: recipes/xrftomo/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  version: {{ version }}
+
+source:
+  url: https://github.com/tomography/{{ name }}/archive/v{{ version }}.tar.gz
+  sha256: 986c4ea85c7fe8bcef274b7085209b74e8a93acd444ad1f9b947fb3f1352cbbb
+
+build:
+  noarch: python
+  number: 0
+  skip: False
```
Comment (carterbox) [inline recipes/xrftomo/meta.yaml]: You can remove `build/skip` if it is `False` always.

**Comment by FabricioSMarin** [comment]
@conda-forge-admin, please restart ci

File: recipes/xrftomo/yum_requirements.txt [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+requirements:
+  build: 
+    - {{ cdt('mesa-libgl-devel') }}  
+    - {{ cdt('mesa-dri-drivers') }}  
+    - {{ cdt('libselinux') }}      
+    - {{ cdt('libxdamage') }} 
+    - {{ cdt('libxxf86vm') }} 
+  host:
+    - xorg-libxfixes  
+
```
Comment (carterbox) [inline recipes/xrftomo/yum_requirements.txt]: This file is not a yaml. Try making the contents literally the following:
```
mesa-libGL
mesa-dri-drivers
libselinux
libXdamage
libXxf86vm
```

**Comment by carterbox** [comment]
@conda-forge-admin, please restart ci

**Comment by carterbox** [comment]
The OSX build was passing before today. It looks like the problem is a conflict between `libtiff` and `libwebp`. `libwebp` was split into two packages since this pull request was opened, so that might be related to the problem. I'm investigating.

**Comment by FabricioSMarin** [comment]
Looks that way. The error mentioned something about incompatible libraries. 
`Reason: Incompatible library version: libtiff.5.dylib requires version 9.0.0 or later, but libwebp.7.dylib provides version 8.0.0`. 
I was looking into it but didnt get very far.

**Comment by carterbox** [comment]
@conda-forge-admin, please reastart ci

**Comment by carterbox** [comment]
@conda-forge/help-python, please review this feedstock. :smile:

File: recipes/xrftomo/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    - scikit-image
+    - tomopy
+
+test:
+  imports:
+    - xrftomo
+
+about:
+  home: https://github.com/tomography/XRFtomo
+  license: BSD-3
```
Comment (npavlovikj) [inline recipes/xrftomo/meta.yaml]: This should be `BSD-3-Clause`.

File: recipes/xrftomo/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  license_family: BSD
+  license_file: LICENSE.txt
+  summary: 'Tomography software for APS beamlines'
+
```
Comment (npavlovikj) [inline recipes/xrftomo/meta.yaml]: This is aesthetic comment, but please remove empty lines within a section.

File: recipes/xrftomo/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  description: |
+    XRFtomo brings together a variety of image processing tools used for preparing, 
+    processing and reconstructing 3D renditions of samples imaged using a 
+    synchrotron-based x-ray fluorescence (XRF) microprobe or nanoprobe at the 
+    Advanced Photon Source (APS).
+
```
Comment (npavlovikj) [inline recipes/xrftomo/meta.yaml]: Please remove empty line.

File: recipes/xrftomo/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+  doc_url: https://xrftomo.readthedocs.io/en/latest/
```
Comment (npavlovikj) [inline recipes/xrftomo/meta.yaml]: This page doesn't exist.

File: recipes/xrftomo/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,60 @@
+
```
Comment (npavlovikj) [inline recipes/xrftomo/meta.yaml]: Please remove empty line.

File: recipes/xrftomo/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+requirements:
+  host:
+    - python >=3
+    - pip
+  run:
+    - dxchange
+    - dxfile
+    - h5py
+    - matplotlib
```
Comment (npavlovikj) [inline recipes/xrftomo/meta.yaml]: I assume you need the `Qt` backend with `matplotlib`?  If you only need to render images to files, then `matplotlib-base` can be used.

File: recipes/xrftomo/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+    processing and reconstructing 3D renditions of samples imaged using a 
+    synchrotron-based x-ray fluorescence (XRF) microprobe or nanoprobe at the 
+    Advanced Photon Source (APS).
+
+  doc_url: https://xrftomo.readthedocs.io/en/latest/
+  dev_url: https://github.com/tomography/XRFtomo
+
+extra:
+  recipe-maintainers:
+    - FabricioSMarin
```
Comment (npavlovikj) [inline recipes/xrftomo/meta.yaml]: @carterbox , do you want to add yourself as a maintainer as well?

Comment (carterbox) [reply recipes/xrftomo/meta.yaml]: No.

File: recipes/xrftomo/meta.yaml (hunk 7 above)
Comment (FabricioSMarin) [reply recipes/xrftomo/meta.yaml]: My mistake, it was removed temporarily until documentation can be completed offline first. I will remove this line for now.

File: recipes/xrftomo/meta.yaml (hunk 9 above)
Comment (FabricioSMarin) [reply recipes/xrftomo/meta.yaml]: The first statement is correct.

**Comment by npavlovikj** [comment]
Thank you for contributing @FabricioSMarin 🎉

**Pull request merged**